spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/PubSubHeaderMapper.java

```diff
@@ -30,16 +30,17 @@
  * Maps headers from {@link com.google.pubsub.v1.PubsubMessage}s to
  * {@link org.springframework.messaging.Message}s and vice-versa.
  *
- * <p>By default, filters out headers called "id" or "timestamp" on the
- * {@link org.springframework.messaging.Message} to {@link com.google.pubsub.v1.PubsubMessage}
+ * <p>By default, filters out headers called "id", "timestamp" or "gcp_pubsub_acknowledgement" on
+ * the {@link org.springframework.messaging.Message} to {@link com.google.pubsub.v1.PubsubMessage}
  * header conversion.
  *
  * @author João André Martins
  */
 public class PubSubHeaderMapper implements HeaderMapper<Map<String, String>> {
 
 	/**
-	 * Patterns of headers to map in {@link #fromHeaders(MessageHeaders, Map)}.
+	 * Patterns of headers to map in {@link #fromHeaders(MessageHeaders, Map)}. First headers take
```

Comment:
Would be better to start a new sentence from a new line.
Also it is really not about *headers*. This is about *patterns*. You can configure it like `gcp_*` and all headers with the `gcp_` prefix will be mapped.